src/irmin.mli

```diff
@@ -1796,75 +1575,625 @@ module Private: sig
+        commit history graph only. *)
+
+    val import: t -> slice -> [`Ok | `Error] Lwt.t
+    (** [import t s] imports the contents of the slice [s] in [t]. Does
+        not modify branches. *)
+
+    val task_of_commit: t -> commit -> task option Lwt.t
+    (** [task_of_commit t c] is the description of the commit
+        [c]. Useful to retrieve the commit date and the committer
+        name. Return [None] if [c] is not present in [t]. *)
+
+  end
+
+  val empty: Repo.t -> t Lwt.t
+  (** [empty repo task] is a temporary, empty store. Becomes a
+      normal temporary store after the first update. *)
+
+  val master: Repo.t -> t Lwt.t
+  (** [master repo] is a persistent store based on [r]'s master
+      branch. This operation is cheap, can be repeated multiple
+      times. *)
+
+  val of_branch: Repo.t -> branch -> t Lwt.t
+  (** [of_branch r name] is a persistent store based on the branch
+      [name]. Similar to [master], but use [name] instead
+      {!Branch.S.master}. *)
+
+  val of_commit: Repo.t -> commit -> t Lwt.t
+  (** [of_commit r c] is a temporary store, based on the commit
+      [c].
+
+      Temporary stores do not have stable names: instead they can be
+      addressed using the hash of the current commit. Temporary stores
+      are similar to Git's detached heads. In a temporary store, all
+      the operations are performed relative to the current head and
+      update operations can modify the current head: the current
+      stores's head will automatically become the new head obtained
+      after performing the update. *)
+
+  val repo: t -> Repo.t
+  (** [repo t] is the repository containing [t]. *)
+
+  val tree: t -> tree Lwt.t
+  (** [tree t] is [t]'s current tree. Contents is not allowed at the
+      root of the tree. *)
+
+  val status: t -> [ `Empty | `Branch of branch | `Commit of commit ]
+  (** [status t] is [t]'s status. It can either be a branch, a commit
+      or empty. *)
+
+  (** Managing the store's heads. *)
+  module Head: sig
+
+    val v: Repo.t -> task -> parents:commit list -> tree -> commit Lwt.t
+    (** [v r task ~parents:p t] is the commit [c] such that:
+        {ul
+        {- [Repo.task_of_commit r c = task]}
+        {- [parents (of_commit r c) = p]}
+        {- [tree (of_commit r c) = t]}}
+    *)
+
+    val list: Repo.t -> commit list Lwt.t
+    (** [list t] is the list of all the heads in local store. Similar
+        to [git rev-list --all]. *)
+
+    val find: t -> commit option Lwt.t
+    (** [find t] is the current head of the store [t]. This works for
+        both persistent and temporary branches. In the case of a
+        persistent branch, this involves getting the the head
+        associated with the branch, so this may block. In the case of
+        a temporary store, it simply returns the current head. Returns
+        [None] if the store has no contents. Similar to [git rev-parse
+        HEAD]. *)
+
+    val get: t -> commit Lwt.t
```

Comment:
Can we get rid of these exception-throwing helpers? It's easy enough to write `Head.find t c >|= crash_if_none` if you really want that. Or maybe rename it to `get_exn` if it provides a useful error message. Otherwise, it's difficult to spot these in code reviews.